specs/electra/beacon-chain.md

```diff
@@ -1076,13 +1076,16 @@ def get_expected_withdrawals(state: BeaconState) -> Tuple[Sequence[Withdrawal],
             ))
             withdrawal_index += WithdrawalIndex(1)
 
-        partial_withdrawals_count += 1
+        processed_partial_withdrawals_count += 1
 
     # Sweep for remaining.
     bound = min(len(state.validators), MAX_VALIDATORS_PER_WITHDRAWALS_SWEEP)
     for _ in range(bound):
         validator = state.validators[validator_index]
-        balance = state.balances[validator_index]
+        # [Modified in Electra:EIP7251]
+        partially_withdrawn_balance = sum(
+            withdrawal.amount for withdrawal in withdrawals if withdrawal.validator_index == validator_index)
```

Comment:
Alternatively, it could iterate over `withdrawals[:effective_partial_withdrawals_count]` which contains partial withdrawals included during the current run. In cases when there are no withdrawal requests eligible for processing this computation will be a noop. But we can also leave the existing computation in the spec for simplicity and defer possible optimisation to the implementations

Reply:
What is `effective_partial_withdrawals_count`? I cannot its definition.

Reply:
And it's not a strong preference, but I do like that better. It would be:

```python
partially_withdrawn_balance = sum(
    withdrawal.amount for withdrawal in withdrawals[:processed_partial_withdrawals_count]
    if withdrawal.validator_index == validator_index
)
```

Reply:
`effective_partial_withdrawals_count = len(withdrawals)` computed before the sweep. `processed_partial_withdrawals_count` may be greater than a number of effective withdrawals due to skipped partial withdrawals

Reply:
That makes sense then.